lua/wikis/commons/Squad/Auto.lua

```diff
@@ -0,0 +1,626 @@
+---@field idleavedate string?
+---@field page string
+---@field name string?
+---@field localizedname string?
+---@field thisTeam SquadAutoTeam
+---@field oldTeam SquadAutoTeam?
+---@field newTeam SquadAutoTeam?
+---@field joindate string
+---@field joindatedisplay string?
+---@field joindateRef table<string, string>?
+---@field leavedate string?
+---@field leavedatedisplay string
+---@field leavedateRef table<string, string>?
+---@field faction string?
+---@field captain boolean?
+
+---@class SquadAutoConfig
+---@field team string
+---@field status SquadStatus
+---@field type SquadType
+---@field title string?
+---@field teams string[]?
+---@field roles {excluded: string[], included: string[]}
+
+---@enum TransferType
+SquadAuto.TransferType = {
+	LEAVE = 'LEAVE',
+	JOIN = 'JOIN',
+	CHANGE = 'CHANGE',
+}
+
+---@class (exact) TeamHistoryEntry
+---@field pagename string
+---@field displayname string
+---@field flag string
+---@field date string
+---@field dateDisplay string
+---@field type TransferType
+---@field references table<string, string>
+---@field wholeTeam boolean
+---@field position string
+---@field fromTeam string?
+---@field fromRole string?
+---@field toTeam string?
+---@field toRole string?
+---@field faction string?
+
+local DEFAULT_INCLUDED_ROLES = {
+    [SquadUtils.SquadType.PLAYER] = {
+        [SquadUtils.SquadStatus.ACTIVE] = {
+            '',
+            'Loan',
+            'Substitute',
+            'Trial',
+            'Stand-in',
+            'Uncontracted'
+        },
+        [SquadUtils.SquadStatus.INACTIVE] = {
+            'Inactive'
+        }
+    },
+    [SquadUtils.SquadType.STAFF] = {},
+}
+
+local DEFAULT_EXCLUDED_ROLES = {
+    [SquadUtils.SquadType.PLAYER] = {},
+    [SquadUtils.SquadType.STAFF] = {
+        '',
+        'Loan',
+        'Substitute',
+        'Trial',
+        'Stand-in',
+        'Uncontracted',
+        'Inactive'
+    },
+}
+
+---Entrypoint for SquadAuto tables
+---@param frame table
+function SquadAuto.run(frame)
+    local autosquad = SquadAuto(frame)
+    autosquad:parseConfig()
+    autosquad:queryTransfers()
+
+    local entries = autosquad:selectEntries()
+    Array.forEach(entries, SquadAuto.enrichEntry)
+
+    return autosquad:display(entries)
+end
+
+---Parses the args into a SquadAutoConfig
+function SquadAuto:parseConfig()
+    local args = self.args
+    local type = SquadUtils.TypeToSquadType[(args.type or ''):lower()]
+    local status = SquadUtils.StatusToSquadStatus[(args.status or ''):lower()]
+    self.config = {
+        team = args.team or mw.title.getCurrentTitle().text,
+        type = type,
+        status = status,
+        title = args.title,
+        roles = {
+            included = Logic.nilIfEmpty(Array.parseCommaSeparatedString(args.roles))
+                or DEFAULT_INCLUDED_ROLES[type][status],
+            excluded = Logic.nilIfEmpty(Array.parseCommaSeparatedString(args.not_roles))
+                or DEFAULT_EXCLUDED_ROLES[type],
```

Comment:
are these args intended to be set on the team pages or did you plan on allowing overwrite via template overwrite?
in the later case imho move as setting to module:Info instead

Reply:
Intended as port of current template overrides, however there's potential to always use default roles.
Will move to Info if that does not suffice

Reply:
#5932